ui/src/app/components/data-user-code-of-conduct.tsx

```diff
@@ -396,176 +396,144 @@ interface Props
-            <DuccSignaturePage
-              {...{ errors, submitting, signatureState, username }}
-              fullName={givenName + ' ' + familyName}
-              signedInitials={duccSignedInitials}
-              signedDate={duccCompletionTimeEpochMillis}
-              onChangeMonitoring={(v) =>
-                this.setState({ initialMonitoring: v })
-              }
-              onChangePublic={(v) => this.setState({ initialPublic: v })}
-              onChangeAccess={(v) => this.setState({ initialAccess: v })}
-              onBack={() =>
-                this.setState({ page: DataUserCodeOfConductPage.CONTENT })
-              }
-              onAccept={() => {
-                this.setState({ submitting: true });
-                // This may record extra GA events if the user views & accepts the DUCC from their profile. If the additional events
-                // are an issue, we may need further changes, possibly disable the Accept button after initial submit.
-                AnalyticsTracker.Registration.AcceptDUCC();
-                wasReferredFromRenewal(this.props.location.search)
-                  ? this.submitCodeOfConductWithRenewal(initialMonitoring)
-                  : this.submitDataUserCodeOfConduct(initialMonitoring);
-                this.setState({ submitting: false });
-              }}
-            />
-          )}
-        </div>
-      );
-    }
-  }
-);
+  const handleAccept = () => {
+    setSubmitting(true);
+    // This may record extra GA events if the user views & accepts the DUCC from their profile. If the additional events
+    // are an issue, we may need further changes, possibly disable the Accept button after initial submit.
+    AnalyticsTracker.Registration.AcceptDUCC();
+    wasReferredFromRenewal(location.search)
+      ? submitCodeOfConductWithRenewal(initialMonitoring)
+      : submitDataUserCodeOfConduct(initialMonitoring);
+    setSubmitting(false);
+  };
+
+  return (
+    <div style={containerStyle}>
+      {signatureState === DuccSignatureState.SIGNED &&
+        !canRenderSignedDucc(duccSignedVersion) && (
+          <SignedVersionFailure duccSignedVersion={duccSignedVersion} />
+        )}
+      {showContentPage && (
+        <DuccContentPage
+          {...{ signatureState }}
+          versionToRender={
+            signatureState === DuccSignatureState.SIGNED
+              ? duccSignedVersion
+              : getLiveDUCCVersion()
+          }
+          buttonDisabled={proceedDisabled}
+          onLastPage={() => setProceedDisabled(false)}
+          onClick={() => setPage(DataUserCodeOfConductPage.SIGNATURE)}
+        />
+      )}
+      {showSignaturePage && (
+        <DuccSignaturePage
+          {...{ errors, submitting, signatureState, username }}
+          fullName={givenName + ' ' + familyName}
+          signedInitials={duccSignedInitials}
+          signedDate={duccCompletionTimeEpochMillis}
+          onChangeMonitoring={(v) => setInitialMonitoring(v)}
```

Comment:
optional but preferred: these can be `onChangeMonitoring={setInitialMonitoring}` now